app/src/main/java/com/palone/planahead/data/database/TaskRepository.kt

```diff
@@ -1,24 +1,15 @@
 package com.palone.planahead.data.database
 
-import com.palone.planahead.data.database.alert.Alert
 import com.palone.planahead.data.database.task.Task
 import kotlinx.coroutines.flow.Flow
 
-class TaskRepository(private val taskAndAlertDao: TaskAndAlertDao) {
-    val allTasks: Flow<Map<Task, List<Alert>>> = taskAndAlertDao.getAllTasksAndAlerts()
-    suspend fun upsertTask(task: Task): Long {
-        return taskAndAlertDao.upsertTask(task)
+class TaskRepository(private val taskDao: TaskDao) {
```

Comment:
The repository and DAO should both have an interface for all common oprations, such as create/update/delete/upsert/find/findall. That way you can have a generic base classes for them and you won't need to do a glue code every time you create a new repository that just calls lower layer.